staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -164,6 +166,19 @@ func (f *structuredMergeManager) Apply(liveObj runtime.Object, patch []byte, man
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to create typed patch object: %v", err)
 	}
+
+	if f.resetFields != nil {
+		resetObjTyped := patchObjTyped.Remove(f.resetFields)
+		diff, err := patchObjTyped.Compare(resetObjTyped)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to compare typed patch to its reset: %v", err)
+		}
+
+		if !diff.IsSame() {
+			return nil, nil, fmt.Errorf("apply contained fields a user should not change:\n%s", diff)
+		}
```

Comment:
this differs from how we handle these fields in other endpoints/methods (e.g. ignore changes under "status" when updating the main resource, ignore changes under "spec" when updating under the status subresource, etc)

it also means that if you take the default serializations of many objects, you will produce an apply patch that will be rejected (most status fields are not pointers, which means they generate at least a `status: {}` stanza when serializing the go types)

Reply:
So you would suggest the same behavior as in update, just strip the fields and proceed?
I already encountered this when i tried to use the decoder when parsing the apply patch instead of yaml.Unmarshal, as i got status: {} this way.
This is probably the reason why yaml.Unmarshal to unstructured was chosen in the first place for apply?

/cc @apelisse @lavalamp 
I think you two discussed this with the suggestion to reject on apply.

Reply:
@liggitt I feel like the second point is much stronger than the first. That's pretty unfortunate :-/

Reply:
> This is probably the reason why yaml.Unmarshal to unstructured was chosen in the first place for apply?

Clearly yes

Reply:
I'll update the code and KEP.